File: Uf-Creator/src/main/java/eu/morphemic/ufcreator/function_creator/function/PredefinedFunction.java
package eu.morphemic.ufcreator.function_creator.function;

import eu.morphemic.ufcreator.function_creator.model.CompositeMetricList;
import eu.morphemic.ufcreator.function_creator.model.ConstantsList;
import eu.morphemic.ufcreator.function_creator.model.RawMetricList;
import eu.morphemic.ufcreator.function_creator.model.VariableList;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor

public abstract class PredefinedFunction {
    public String name;
    public List<VariableList> variableList;
    public List<ConstantsList> constantsList;
    public List<RawMetricList> rawMetricList;
    public List<CompositeMetricList> compositeMetricList;
    public double weight;

}

File: UfCreatorUI/src/app/model/PredefinedFunction.ts
import {Variable} from "./Variable";
import {Constant} from "./Constant";
import {Metric} from "./Metric";

export class PredefinedFunction {
  name: string
  src: string
  variableList: Variable[]
  constantsList: Constant[]
  metricList: Metric[]
  weight: number

  constructor(name: string, src: string, variableList: Variable[], constantsList: Constant[], metricList: Metric[]) {
    this.name = name;
    this.src = src;
    this.variableList = variableList;
    this.constantsList = constantsList;
    this.metricList = metricList;
  }
}
